Maple2.File.IO/Nif/NifBlock.cs

```diff
@@ -0,0 +1,23 @@
+﻿namespace Maple2.File.IO.Nif;
+
+public class NifBlock {
+    public string BlockType { get; init; }
+    public bool HasName { get; init; }
+    public int BlockIndex { get; init; }
+    public string Name;
+
+    public NifBlock(string blockType, bool hasName, int blockIndex) {
+        BlockType = blockType;
+        HasName = hasName;
+        BlockIndex = blockIndex;
+        Name = string.Empty;
+    }
+
+    public virtual void Parse(NifDocument document) {
+        if (HasName) {
+            Name = document.ReadString();
+        }
+
+        document.Reading(this);
```

Comment:
why not directly call `document.Parse()`? is it because you need to defer execution?

Reply:
That was more to track which object was currently being read. I can just simplify the code & move that to document.Parse() directly